extensions/rich_text_components/Video/Video.js

```diff
@@ -25,16 +25,44 @@ oppia.directive('oppiaNoninteractiveVideo', [
       restrict: 'E',
       scope: {},
       templateUrl: 'richTextComponent/Video',
-      controller: ['$scope', '$attrs', 'explorationContextService',
-        function($scope, $attrs, explorationContextService) {
+      controller: ['$scope', '$attrs', 'explorationContextService', '$element',
+        'PAGE_CONTEXT',
+        function($scope, $attrs, explorationContextService, $element) {
           var start = oppiaHtmlEscaper.escapedJsonToObj($attrs.startWithValue);
           var end = oppiaHtmlEscaper.escapedJsonToObj($attrs.endWithValue);
 
-          $scope.autoplaySuffix = (oppiaHtmlEscaper.escapedJsonToObj(
-            $attrs.autoplayWithValue) ? '&autoplay=1' : '&autoplay=0');
           $scope.videoId = oppiaHtmlEscaper.escapedJsonToObj(
             $attrs.videoIdWithValue);
           $scope.timingParams = '&start=' + start + '&end=' + end;
+
+          // This code helps in checking whether video is for upcoming content.
+          // This is totally dependent of parent div of video container
+          // If in future that parent div changes then this code will be
+          // required to change accordingly.
+          var isVisible = true;
+          var videoVisibilty = angular.element($element).prop('offsetParent');
+          if (videoVisibilty) {
+            isVisible = (videoVisibilty.getAttribute('angular-html-bind') !==
+              'upcomingContentHtml' ? true : false);
+          }
+
+          // If user in learner context then only autoplay videos.
+          if (explorationContextService.getPageContext() ===
+            PAGE_CONTEXT.LEARNER) {
+            // If video has been already played or if it is not visible for now
+            // then dont autoplay it.
+            if (explorationContextService.getFromPlayedList($scope.videoId) ||
+              !(isVisible)) {
+              $scope.autoplaySuffix = '&autoplay=0';
+            } else {
+              $scope.autoplaySuffix = '&autoplay=1';
+            }
+          } else {
+            // If not in learner context then use default value,
```

Comment:
Ah -- there's been a miscommunication here, sorry. All the stuff about autoplaying should only apply if the creator has said "autoplay this video" in the first place, otherwise the behavior should be to just not autoplay. Does that make sense? It would be weird for us to autoplay videos if the creator has already said "do not autoplay".